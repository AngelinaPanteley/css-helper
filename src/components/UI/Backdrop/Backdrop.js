import React from 'react';
import PropTypes from 'prop-types';
import styles from './Backdrop.scss';

const backdrop = (props) => {
  const classes = [styles.Backdrop];
  if (props.withBackground) {
    classes.push(styles.With_Background);
  }
  return (
    props.show
      ? <div className={classes.join(' ')} onClick={props.clicked}></div>
      : null
  )
};

backdrop.propTypes = {
  withBackground: PropTypes.bool,
  clicked: PropTypes.func.isRequired,
}

export default backdrop;