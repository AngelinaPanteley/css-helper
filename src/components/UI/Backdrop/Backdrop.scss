@import '../../../variables';

.Backdrop {
  width: 100%;
  height: 100%;
  position: fixed;
  z-index: map-get($z-index, backdrop);
  left: 0;
  top: 0;

  &.With_Background {
    background-color: $color-backdrop-bg;
  }
}