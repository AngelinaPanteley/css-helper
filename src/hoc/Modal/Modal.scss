@import '../../variables';

.Modal {
  @include xy-center;

  position: absolute;
  width: auto;
  max-width: 70%;
  height: auto;
  z-index: map-get($z-index, popup);
  background-color: $color-gray;
  box-sizing: border-box;
  transition: transform 0.3s ease-out;
  padding: 40px;
  border-radius: 5px;
  top: 30vh;

  &.Open {
    transform: translate(-50%, 0);
  }

  &.Close {
    transform: translate(-50%, calc(-100% - 30vh));
  }
}