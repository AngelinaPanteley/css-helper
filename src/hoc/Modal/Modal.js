import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from './Modal.scss';
import Backdrop from '../../components/UI/Backdrop/Backdrop';
import Auxiliary from '../Auxiliary/Auxiliary';

class Modal extends PureComponent {
  static propTypes = {
    isOpen: PropTypes.bool.isRequired,
    onClose: PropTypes.func.isRequired,
  }

  render() {
    const attachedClasses = [styles.Modal];
    if (this.props.isOpen) {
      attachedClasses.push(styles.Open);
    } else {
      attachedClasses.push(styles.Close);
    }
    return (
      <Auxiliary>
        <Backdrop
          show={this.props.isOpen}
          clicked={this.props.onClose}
          withBackground={true} />
        <div className={attachedClasses.join(' ')}>
          {this.props.children}
        </div>
      </Auxiliary>
    );
  }
}

export default Modal;