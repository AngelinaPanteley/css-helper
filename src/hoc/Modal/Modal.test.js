"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import Modal from './Modal.js';
import Backdrop from '../../components/UI/Backdrop/Backdrop.js';

configure({ adapter: new Adapter() });

describe('>>>Tabs', () => {

  //SNAPSHOTS

  const props = {
    isOpen: true,
    onClose: () => { },
  }

  const modalComponent = <Modal {...props}>
    <div>Child</div>
  </Modal>;

  const component = renderer.create(modalComponent);

  it('renders itself correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  //SHALLOW RENDERING

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(modalComponent);
  });

  it('renders Backdrop', () => {
    chaiExpect(wrapper.find(Backdrop)).to.have.lengthOf(1);
  });
})