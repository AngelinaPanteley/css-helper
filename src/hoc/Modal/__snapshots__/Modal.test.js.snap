// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Tabs renders itself correctly 1`] = `
Array [
  <div
    className=" "
    onClick={[Function]}
  />,
  <div
    className=" "
  >
    <div>
      Child
    </div>
  </div>,
]
`;
